atom/renderer/api/atom_api_web_frame.cc

```diff
@@ -231,11 +249,13 @@ void WebFrame::SetSpellCheckProvider(mate::Arguments* args,
       std::make_unique<SpellCheckClient>(language, args->isolate(), provider);
   // Set spellchecker for all live frames in the same process or
   // in the sandbox mode for all live sub frames to this WebFrame.
-  FrameSpellChecker spell_checker(
-      client.get(), content::RenderFrame::FromWebFrame(web_frame_));
+  auto* render_frame = content::RenderFrame::FromWebFrame(web_frame_);
+  FrameSpellChecker spell_checker(client.get(), render_frame);
   content::RenderFrame::ForEach(&spell_checker);
   spell_check_client_.swap(client);
   web_frame_->SetSpellCheckPanelHostClient(spell_check_client_.get());
+  web_frame_observer_.reset(
```

Comment:
This can be just `new AtomWebFrameObserver(render_frame, std::move(spell_check_client_)` instead of storing it in a member variable, frame observers are meant to be self destructing classes since the observed frame can go away at any time signaled by `OnDestruct`. There is no point in holding onto the observer after that.